### ISSUE TITLE
Memory issues

### ISSUE THREAD

**Issue by piranna** [NodeOS/NodeOS#179]
I've just made `logon` to use [kexec](https://github.com/jprichardson/node-kexec) and I've been able to see a decrement of memory since we now have one Node.js instance less on memory... on only about 1mb :-/ Seems the memory problem is not specially about the build-ins because seems the kernel is being intelligent and reusing automatically the memory space of the Node.js binary (I find 1mb to be ok for the instance stack and heap), so the problem should be in other place. The only thing I can think about wasting so much memory is about the initram, because it has the Node.js binary all the time in memory, but it's only about 10mb... where could be the rest of the memory being wasted? :-(

![pantallazo-qemu](https://cloud.githubusercontent.com/assets/532414/11180446/1f4ed980-8c5b-11e5-8fb0-2f7e058a558b.png)

**Comment by piranna** [comment]
Garbage collector usage [needs to be reviewed](http://blog.heroku.com/archives/2015/11/10/node-habits-2016#8-hook-things-up).

**Comment by lite20** [comment]
What's the status of this?  Has this been corrected?  Is this a zombie thread not being cleaned up? :P

**Comment by piranna** [comment]
No, this is still on focus. Seems memory is a problem since each v8 instance consumes about 10mb, so it's needed to investigate both how to make it consume less memory and also how to prevent out-of-memory errors. We now know the problem when running npm or slap was due 128mb was too few for NodeOS, but on the other hand I would love to see it working on more memory constrained systems or also my old XBox with 64mb :-P It also needs the garbage collector to be tune-up so it's more agressive, for example by default it takes an upper limit of 1.5GBs, that on systems with less memory is fairly useless... :-/

**Comment by lite20** [comment]
bumping this thread because I don't believe this has been resolved. I'll look into making v8 more aggressive and use less ram.
